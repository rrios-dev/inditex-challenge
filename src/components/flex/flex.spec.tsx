import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import Flex, { FlexProps } from ".";

const items = Array.from({ length: 10 }).map((_, index) => (
  <Flex key={index}>{index}</Flex>
));

describe("Flex component", () => {
  describe("Direction", () => {
    it.each(["row", "column"] as const)("%s", (direction) => {
      const { asFragment } = render(<Flex direction={direction}>{items}</Flex>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("Justify content", () => {
    it.each(["start", "center", "end", "between", "around"] as const)(
      "%s",
      (justify) => {
        const { asFragment } = render(<Flex justify={justify}>{items}</Flex>);
        expect(asFragment()).toMatchSnapshot();
      }
    );
  });

  describe("Align Items", () => {
    it.each(["start", "center", "end", "stretch"] as const)("%s", (align) => {
      const { asFragment } = render(<Flex align={align}>{items}</Flex>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("Wrap", () => {
    it.each(["wrap", "nowrap"] as const)("%s", (wrap) => {
      const { asFragment } = render(<Flex wrap={wrap}>{items}</Flex>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("Gap", () => {
    Array.from({ length: 12 }).forEach((_, index) => {
      it((index + 1).toString(), () => {
        const { asFragment } = render(
          <Flex gap={(index + 1) as FlexProps<any>["gap"]}>{items}</Flex>
        );
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
