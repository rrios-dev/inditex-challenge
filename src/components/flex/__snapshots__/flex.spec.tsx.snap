// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flex component > Align Items > center 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _align-center_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Align Items > end 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _align-end_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Align Items > start 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _align-start_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Align Items > stretch 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _align-stretch_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Direction > column 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _dir-column_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Direction > row 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _dir-row_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 1 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-1_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 2 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-2_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 3 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-3_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 4 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-4_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 5 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-5_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 6 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-6_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 7 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-7_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 8 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-8_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 9 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-9_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 10 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-10_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 11 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-11_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Gap > 12 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _gap-12_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Justify content > around 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _justify-around_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Justify content > between 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _justify-between_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Justify content > center 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _justify-center_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Justify content > end 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _justify-end_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Justify content > start 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _justify-start_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Wrap > nowrap 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _wrap-nowrap_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;

exports[`Flex component > Wrap > wrap 1`] = `
<DocumentFragment>
  <div
    class="_flex_6eb20e _wrap-wrap_6eb20e"
  >
    <div
      class="_flex_6eb20e"
    >
      0
    </div>
    <div
      class="_flex_6eb20e"
    >
      1
    </div>
    <div
      class="_flex_6eb20e"
    >
      2
    </div>
    <div
      class="_flex_6eb20e"
    >
      3
    </div>
    <div
      class="_flex_6eb20e"
    >
      4
    </div>
    <div
      class="_flex_6eb20e"
    >
      5
    </div>
    <div
      class="_flex_6eb20e"
    >
      6
    </div>
    <div
      class="_flex_6eb20e"
    >
      7
    </div>
    <div
      class="_flex_6eb20e"
    >
      8
    </div>
    <div
      class="_flex_6eb20e"
    >
      9
    </div>
  </div>
</DocumentFragment>
`;
